rules/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector.php

```diff
@@ -88,7 +89,13 @@ public function refactor(Node $node): ?Node
                 continue;
             }
 
-            if (! $method->isPrivate()) {
+            $isPrivate =
+                ($node->isFinal()
+                && $node->extends === null
```

Comment:
`$node->implements` empty should be checked as well, also probably used by/from trait.

service from Parent override guard or FamilyRelation services may be can be used for this case

Reply:
good point. added a test-case and fixed it

Reply:
I did not use the suggested services, as I think pure AST based decision are more reliable

Reply:
Ok, lets use this as initial support and then we can improve later with verify parent FullyQualified->toString() is autoloaded